### ISSUE TITLE
Rare rendering issue with `path()`

### ISSUE THREAD

**Issue by Mc-Zen** [typst/typst#5267]
### Description

There exists a weird rendering issue with the following path
```typ
#std.path(
  ((40pt, 10pt), (0pt, -12pt)),
  (7.4906425pt, 20.41pt)
)
```

![image](https://github.com/user-attachments/assets/570bcc95-b2a4-4e13-9e85-ce5b883c135a)

Note that this is a very normal path with no lines going back into themselves or anything. The issue goes away when replacing `7.4906425pt` with `7.4906421pt`. 

This happens both with SVG and PNG export, but not with PDF. 

It is not a regression, the behaviour has been just the same in Typst 0.6.0. 

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
@TS53 I don't see why typst should bend over backwards to  fix SVG renderers that do not follow the spec

**Comment by TS53** [comment]
It seems that this is the only difference between the two generated svg files. Also, when all other floating point numbers are converted to integers, the bug stays. Maybe, the path drawing only allows 6 digits after the dot. I don't know.

I have no idea about SVG, but for me the SVG path definition sound like ghostscript operators and I'm pretty sure they neved allowed so many digits.

**Comment by TS53** [comment]
You are right, it seems other f32 values with 7 digits after the dot work like 7.4999995

But if it turns out that only numbers with 7 digits make problems, it would be a simple solution to just make the numbers shorter.
I also think, it would be best to pass the same numbers to all formats.

**Comment by TS53** [comment]
Hm, I think I have now found the buggy pattern:

7.4999995 works when written directly into the svg. When written into typst, it will become 7.5
The browsers SVG render probably uses the same algo like typst, takes the 7.4999995 and converts it into 7.5 without telling the user.

But: under the assumption that they use the same algo, 7.4906425 stays from the typst side 7.4906425 and thus also on the browsers side. Thus it stays 7 digit and problematic.

This is my guess. As said, making the number shorter will probably solves this.
